packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -57,7 +57,7 @@ export async function generateSideloadFile(app: OfficeApp, manifest: ManifestInf
   const templatePath = document && document !== "" ? path.resolve(document) : getTemplatePath(app, addInType);
 
   if (!templatePath) {
-    throw new ExpectedError("Sideload is not supported.");
+    throw new ExpectedError(`Sideload is not supported for apptype: ${addInType}.`);
```

Comment:
Was it just the apptype we were looking at?  Are there some other pieces of information that would help us better understand the code path that was taken?

Also, do we just have access to the error message or can we add information to the property bags we were looking at?

Reply:
We shouldn't add addInType to the message here. It's not useful info for the customer. There are 3 add in types, TaskPane, Content, and MailApp. It's not really the add-in type which determines if sideload is supported or not. 

What is the scenario where the templatePath is not being returned from getTemplatePath?

Reply:
I think the apptype for this specific error should be enough right? The main problem with this error is that it didn't give enough information about the error (which is this case).

What do you mean by property bags? It is something like additional properties? It is possible to add these when the report errors function is called.

Reply:
Igor, see my previous comment. It's not the app type which is the issue here but more of a question of why getTemplate didn't return a string.

Reply:
Igor, if you remember our working session there were filed called "customDimensions" and "details" that contained data like the ExpectedError boolean.  Instead of putting these values in the error message (unless they are also helpful to the user) they should be in data fields like that.

This error gets logged when getTemplatePath doesn't return a path.  That method uses a combination of OfficeApp and AddInType to return the path and there is not default response.  It also looks like the matrix of cases small (includes just three hosts and two types).  Logging both of these values would help us understand what cases are coming back as undefined triggering the error.

In the last 7 day this is by far our top error (with hundreds of hits) and it's too generic to determine what is causing it.  Determining that would lead us to understand what cases lead the customers to hit that error and consider if there is something we can do to improve their experience.  I'm not sure if the same information being put in the displayed error message would be helpful to the customers or not.

Reply:
I should be able to add those fields to the error details, it is just on another part of the code. In this case, it is in the `startDebugging` function inside office-addin-debugging.

But we should make the error that the user see better right? If it is not the apptype, which more information should we give to the user?

Reply:
@millerds I just added the new dimensions to the reported error message: app, apptype and document.

Reply:
@akrantz Could you suggest a better message to be sent to the user? It needs to help him identify the cause of the issue and the current one is bad at that (I say bad because me and Darren spend a good amount of time trying to figure it out).